### PR TITLE
Updates from previous PR

### PR DESCRIPTION
**Pull request by RajeshwariJadhav**
From the previous **popup.js** PR, some changes were requested after PR was closed. Those changes have been made in this PR. 

- URL for appsettings without u/\<num\> were added. Example: https://admin.google.com/ac/appsettings/725740718362/videoSettings is now a valid URL 
- Test case for this new URL form was added, and tests were passed. 
image

- The _findPagecase_ function was renamed to _findPageType_ in all files. 

- Dynamically create and remove mock buttons from the DOM for testing using _createElement_ and _remove()_ in the _beforeEach_ and _afterEach_ sections of Jasmine testing.

### CODE REVIEW

File: STEP_Project/popup.js [hunk 1]
```diff
@@ -24,12 +24,13 @@ function findPagecase(givenurl)
   let orgunitspage = givenurl.match(/admin.google.com\/u\/[0-9]+\/ac\/orgunits/g);
   let settingspage = givenurl.match(/admin.google.com\/u\/[0-9]+\/ac\/appsettings\/[0-9]+\/.+/g);
   let onoffpage = givenurl.match(/admin.google.com\/u\/[0-9]+\/ac\/settings\/serviceonoff.*/g);
+  let firstsettingspage = givenurl.match(/admin.google.com\/ac\/appsettings\/[0-9]+\/.+/g);
```
Comment (atishey-jain) [inline STEP_Project/popup.js:27]: This change will be needed for all pages, orgunits, settings and onoff. Suggest you to not create a new variable but modify the existing regexes to optionally incorporate this as you did in pageUrl above.

File: STEP_Project/tests/popupTest.js [hunk 2]
```diff
@@ -37,21 +42,20 @@ describe('Testing enabling and disabling of buttons depending on pageType', ()=>
   beforeEach(function() 
   {
     //back to default for future tests
+    const arButton = document.createElement('BUTTON')
+    arButton.setAttribute("id", "addRemove");
+    const sButton = document.createElement('BUTTON')
+    sButton.setAttribute("id", "select");
+    document.body.appendChild(arButton);
+    document.body.appendChild(sButton);
```
Comment (atishey-jain) [inline STEP_Project/tests/popupTest.js:50]: Move this to a separate function.

File: STEP_Project/tests/popupTest.js [hunk 3]
```diff
@@ -37,21 +42,20 @@ describe('Testing enabling and disabling of buttons depending on pageType', ()=>
...
-    if (s !== null)
-    {
-      s.disabled = false;
-    }
-    if (ar !== null)
-    {
-      ar.disabled = false;
-    }
+    document.getElementById("select").remove();
+    document.getElementById("addRemove").remove();
```
Comment (atishey-jain) [inline STEP_Project/tests/popupTest.js:58]: Maybe move this to a separate function and call that function here.

File: STEP_Project/popup.js (hunk 1 above)
Comment (RajeshwariJadhav) [reply STEP_Project/popup.js]: done, regex expressions were modified and extra variable was removed.

File: STEP_Project/tests/popupTest.js (hunk 2 above)
Comment (RajeshwariJadhav) [reply STEP_Project/tests/popupTest.js]: Moved to _initTest_ function

File: STEP_Project/tests/popupTest.js (hunk 3 above)
Comment (RajeshwariJadhav) [reply STEP_Project/tests/popupTest.js]: moved to _restoreDOM_ function

File: STEP_Project/tests/popupTest.js [hunk 4]
```diff
@@ -36,26 +36,33 @@ describe('Testing pageType that determines enabling/disabling of buttons dependi
...
  * testing function disableButtons
  */
+function initTest()
+{
+  const arButton = document.createElement('BUTTON')
+  arButton.setAttribute("id", "addRemove");
+  const sButton = document.createElement('BUTTON')
+  sButton.setAttribute("id", "select");
+  document.body.appendChild(arButton);
+  document.body.appendChild(sButton);
```
Comment (atishey-jain) [inline STEP_Project/tests/popupTest.js:46]: Suggestion was to move only this part in a separate function, let's say createDom (as you did for removing DOM elements). We can keep the initialization of select and remove buttons in the beforeEach method itself.

File: STEP_Project/tests/popupTest.js [hunk 5]
```diff
@@ -36,26 +36,33 @@ describe('Testing pageType that determines enabling/disabling of buttons dependi
...
+  s = document.getElementById("select");
```
Comment (atishey-jain) [inline STEP_Project/tests/popupTest.js:47]: Rename this to something like selectButton and similar for "ar" variable.

File: STEP_Project/tests/popupTest.js (hunk 4 above)
Comment (RajeshwariJadhav) [reply STEP_Project/tests/popupTest.js]: updated function modularization

File: STEP_Project/tests/popupTest.js (hunk 5 above)
Comment (RajeshwariJadhav) [reply STEP_Project/tests/popupTest.js]: variables renamed

**Pull request merged**